### PR TITLE
Allow assigning of licenses over assigned quota

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1505

### CODE REVIEW

File: inc/computer_softwarelicense.class.php [hunk 1]
```diff
@@ -373,7 +373,8 @@ static function showForLicense(SoftwareLicense $license) {
       //If the number of linked assets have reached the number defined in the license,
```
Comment (orthagh) [inline inc/computer_softwarelicense.class.php]: update the comment

File: install/update_94_100.php [hunk 2]
```diff
@@ -41,15 +42,21 @@ function update94to100() {
    $current_config   = Config::getConfigurationValues('core');
    $updateresult     = true;
    $ADDTODISPLAYPREF = [];
+   $config_to_drop = [];
 
    //TRANS: %s is the number of new version
-   $migration->displayTitle(sprintf(__('Update to %s'), '10.0.0'));
-   $migration->setVersion('10.0.0');
+   $migration->displayTitle(sprintf(__('Update to %s'), '10.0'));
+   $migration->setVersion('10.0');
```
Comment (orthagh) [inline install/update_94_100.php]: not sure about this, ping @trasher

Comment (trasher) [reply install/update_94_100.php]: Version must be `10.0.0`, yep

File: install/update_94_100.php [hunk 3]
```diff
@@ -41,12 +42,18 @@ function update94to100() {
...
+   $config_to_drop = [];
 
    //TRANS: %s is the number of new version
    $migration->displayTitle(sprintf(__('Update to %s'), '10.0.0'));
    $migration->setVersion('10.0.0');
 
-   //put your update rules here, and drop the line!
+   //Add over-quota option to software licenses to allow assignment after all alloted licenses are used
+   if (!$DB->fieldExists('glpi_softwarelicenses', 'allow_overquota')) {
+      if ($migration->addField('glpi_softwarelicenses', 'allow_overquota', "tinyint(1) NOT NULL DEFAULT '0'")) {
```
Comment (trasher) [inline install/update_94_100.php]: Probably better to rely on "bool" migration type which does the same

**Pull request merged**